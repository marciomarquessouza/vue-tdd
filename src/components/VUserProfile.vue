<template>
  <div>
    <h1>{{ user.name }}</h1>
  </div>
</template>

<script>
export default {
  props: {
    user: {
      required: true,
      type: Object,
      default: () => ({}),
    },
  },
};
</script>
