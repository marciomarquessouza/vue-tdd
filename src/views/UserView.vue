<template>
  <div>
    <h1>Hello World</h1>
    <user-search />
    <user-profile />
  </div>
</template>

<script>

import VUserProfile from '@/components/VUserProfile';
import VUserSearch from '@/components/VUserSearch';

export default {
  name: 'UserView',
  components: {
    'user-profile': VUserProfile,
    'user-search': VUserSearch,
  },
};
</script>
