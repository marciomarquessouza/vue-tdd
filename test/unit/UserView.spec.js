import { shallowMount } from '@vue/test-utils';
import UserView from '@/views/UserView';
import VUserSearch from '@/components/VUserSearch';
import VUserProfile from '@/components/VUserProfile';

describe('UserView', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(UserView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders main childs', () => {
    const wrapper = shallowMount(UserView);
    const UserSearch = wrapper.find(VUserSearch);
    const UserProfile = wrapper.find(VUserProfile);
    // Assets
    expect(UserSearch.exists()).toBe(true);
    expect(UserProfile.exists()).toBe(true);
  });
});
