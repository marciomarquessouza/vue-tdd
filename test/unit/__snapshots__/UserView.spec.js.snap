// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders correctly 1`] = `
<div>
    <h1>Hello World</h1>
    <user-search-stub></user-search-stub>
    <user-profile-stub></user-profile-stub>
</div>
`;
